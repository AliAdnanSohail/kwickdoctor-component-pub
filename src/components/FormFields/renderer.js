import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
