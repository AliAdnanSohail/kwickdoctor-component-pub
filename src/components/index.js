export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
