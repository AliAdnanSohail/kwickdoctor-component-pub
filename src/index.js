export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
