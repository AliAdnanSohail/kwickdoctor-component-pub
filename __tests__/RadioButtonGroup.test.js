import React from 'react';
import renderer from 'react-test-renderer';

import { RadioButtonGroup } from '../src';

const defaultProps = {
  id: 'gender',
  label: 'Label',
  name: 'gender',
  options: [{ value: 'male' }, { value: 'female' }],
};

it('renders correctly', () => {
  const element = renderer.create(<RadioButtonGroup {...defaultProps} />).toJSON();

  expect(element).toMatchSnapshot();
});
