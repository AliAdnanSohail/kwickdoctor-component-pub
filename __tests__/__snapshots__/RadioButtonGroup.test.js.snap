// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 form-field"
>
  <fieldset
    className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 fieldset"
  >
    <legend
      className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 legend"
    >
      Label
    </legend>
    <div
      className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 radio-buttons"
    >
      <label
        className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 radio-button"
      >
        male
        <input
          className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 "
          defaultChecked={false}
          name="gender"
          type="radio"
          value="male"
        />
      </label>
      <label
        className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 radio-button"
      >
        female
        <input
          className="jsx-1685083184 jsx-4038665278 jsx-2062352977 jsx-904933394 jsx-2140168082 "
          defaultChecked={false}
          name="gender"
          type="radio"
          value="female"
        />
      </label>
    </div>
  </fieldset>
</div>
`;
